woocommerce/woocommerce
# Blueprint - Disregard invalid slug and continue

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR addresses an edge case where the zip export fails on a JN site due to the directory name used for WC. Although this isn't a bug, I'm fixing it here since we frequently test on JN sites. We'll have a better structure for handling exception flows after the beta testing period.


On JN sites, WC directory name has "wc_beta_tester_live_branch_9.3.0-dev-10360962882-g0468bdb" format. 

Error:

```
Uncaught InvalidArgumentException: Invalid plugin slug: wc_beta_tester_live_branch_9.3.0-dev-10360962882-g0468bdb in /srv/htdocs/wp-content/plugins/wc_beta_tester_live_branch_9.3.0-dev-10360962882-g0468bdb/vendor/woocommerce/blueprint/src/ZipExportedSchema.php:183
```



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Skip the core profiler and enable `blueprint` feature flag.
3. Go to `WooCommerce -> Settings -> Advanced -> Blueprint`
4. Check on the `Export as a zip (Experimental)`
5. Click on the `Export` button.
6. Confirm export works without an error.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

This is a bug fix specific for JN sites and the blueprint is behind a feature flag for beta testing. 

</details>
